__tests__/utils.test.js

```diff
@@ -13,6 +14,11 @@ describe('unit tests for utility functions', () => {
     const input = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
     const shuffled = shuffleDeck(input);
     expect(shuffled.length).toBe(10);
+    // shuffled[0] could be one, given that the randomIdx is truly random
+    // so this test may fail sometimes
+    // you will likely need to mock Math.random in order for this test to not fail periodically
+    // In a real application, this might fail mysteriously when the code is deployed
+    // Tests should be completely deterministic.  
```

Comment:
I realized that but don't know how to avoid it. How would I mock Math.random?

Reply:
you can do something like this in your before / before each
```
	global.Math.random = (() => {
		const randomNumbers = [0.1, 0.9, 0.2, 0.8, 0.3, 0.7, 0.4, 0.6, 0.5];
		let t = -1;
		return () => {
			t = (t + 1) % randomNumbers.length;
			return randomNumbers[t];
		};
	})();
```
and then put it back on your after each... you'll need to store the original math.random in a temporary variable

or try this?
https://www.npmjs.com/package/jest-mock-random